internal/proxy/oauthproxy.go

```diff
@@ -781,25 +829,6 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) (er
 		}
 	}
 
-	// We revalidate group membership whenever the session is refreshed or revalidated
```

Comment:
I'm on board with not having to run these validators on every request. However, there is a part of me that's anxious about removing these checks.

What validation do we do to authenticate and authorize this request to this upstream beyond these checks? I want to make sure that at least it's impossible to take a valid token from upstream foo and use it against upstream bar.